airflow/hooks/S3_hook.py

```diff
@@ -105,7 +105,9 @@ def __init__(
         self._creds_in_conn = 'aws_secret_access_key' in self.extra_params
         self._creds_in_config_file = 's3_config_file' in self.extra_params
         self._default_to_boto = False
-        if 'host' in self.extra_params:
+        if self.s3_conn.host is not None:
+            self.s3_host = self.s3_conn.host
+        elif 'host' in self.extra_params:
```

Comment:
I'm guessing this is to support S3 compatible services etc?

Can we add an something to the docs (somewhere? Not sure where) mentioning that this is supported? And also probably add a deprecation warning if `host' in self.extra_params` - pulling that from the connection's host field makes much more sense.

Reply:
I agree this is quite useful for S3 compatible services, for example IBM's COS (Cloud Object Storage, my team is using it).

Currently we need to set value 'host' in `extra_params`, which is not documented clearly. I only figured this out after reading source code.